src/main/java/org/radarbase/appserver/service/RadarProjectService.java

```diff
@@ -1,15 +1,135 @@
+ *  *
+ *  *  * Copyright 2018 King's College London
+ *  *  *
+ *  *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  * you may not use this file except in compliance with the License.
+ *  *  * You may obtain a copy of the License at
+ *  *  *
+ *  *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *  * See the License for the specific language governing permissions and
+ *  *  * limitations under the License.
+ *  *  *
+ *  *
+ *
+ */
+
 package org.radarbase.appserver.service;
 
+import org.radarbase.appserver.converter.*;
 import org.radarbase.appserver.dto.RadarProjectDto;
 import org.radarbase.appserver.dto.RadarProjects;
+import org.radarbase.appserver.entity.Project;
+import org.radarbase.appserver.exception.InvalidProjectDetailsException;
+import org.radarbase.appserver.exception.NotFoundException;
+import org.radarbase.appserver.repository.ProjectRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
 
+/**
+ * {@link Service} for interacting with the {@link Project} {@link javax.persistence.Entity} using the
+ * {@link ProjectRepository}.
+ *
+ * @author yatharthranjan
+ */
+@Service
 public class RadarProjectService {
 
+    @Autowired
+    private ProjectRepository projectRepository;
+
+    @Transactional(readOnly = true)
     public RadarProjects getAllProjects() {
-        return null;
+        RadarProjects projects = new RadarProjects();
+        return new RadarProjects().setProjects(
+                ConverterFactory.getProjectConverter().entitiesToDtos(projectRepository.findAll()));
+    }
+
+    @Transactional(readOnly = true)
+    public RadarProjectDto getProjectById(Long id) {
+        Optional<Project> project = projectRepository.findById(id);
+
+        if (project.isPresent()) {
+            return ConverterFactory.getProjectConverter().entityToDto(project.get());
+        } else {
+            throw new NotFoundException("Project not found with id" + id);
+        }
     }
 
-    public RadarProjectDto getProjectById(String id) {
-        return null;
+    @Transactional(readOnly = true)
+    public RadarProjectDto getProjectByProjectId(String projectId) {
+        Optional<Project> project = projectRepository.findByProjectId(projectId);
+
+        if (project.isPresent()) {
+            return ConverterFactory.getProjectConverter().entityToDto(project.get());
+        } else {
+            throw new NotFoundException("Project not found with projectId" + projectId);
+        }
+    }
+
+    @Transactional(readOnly = true)
+    public Project getProjectEntityByProjectId(String projectId) {
```

Comment:
Not sure why both of these are necessary. In any case, you can use the second in the former then convert to DTo, to avoid code duplication :)

Reply:
Thanks.